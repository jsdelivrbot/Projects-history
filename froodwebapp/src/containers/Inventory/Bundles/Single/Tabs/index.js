export ConnectedMainTab from './MainTab/MainTab';
export ConnectedItemsTab from './ItemsTab/ItemsTab';
export ConnectedAssemblyTab from './AssemblyTab/AssemblyTab';
export Tabs from './Tabs';
