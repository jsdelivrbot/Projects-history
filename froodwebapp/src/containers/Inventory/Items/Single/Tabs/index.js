export ConnectedMainTab from './MainTab/MainTab';
export ConnectedBinsTab from './BinsTab/BinsTab';
export ConnectedSuppliersTab from './SuppliersTab/SuppliersTab';
export ConnectedUOMTab from './UOMTab/UOMTab';
export StockLifeCycleTab from './StockLifeCycleTab/StockLifeCycleTab';
export ImagesTab from './ImagesTab/ImagesTab';
export Tabs from './Tabs';
