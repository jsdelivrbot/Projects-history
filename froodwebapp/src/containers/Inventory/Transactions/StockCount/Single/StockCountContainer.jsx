/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ConnectedStockCount from './StockCount/StockCount';

const mapStateToProps = state => ({
  loadingPage: state.priceList.loadingPage,
  priceList: state.priceList.priceList,
  login: state.login.user,
});

export class StockCountContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      stockCountId: this.props.match.params.id,
      isNewStockCount: this.props.match.params.id === 'new'
    };
  }

  componentWillReceiveProps(nextProps) {
    const stockCountId = nextProps.match.params.id;

    this.setState({
      stockCountId,
      isNewStockCount: stockCountId === 'new'
    });
  }


  render() {
    const {
      stockCountId,
      isNewStockCount
    } = this.state;

    return (
      <ConnectedStockCount
        stockCountId={ Number(stockCountId) }
        isNewStockCount={ isNewStockCount }
      />
    );
  }
}

StockCountContainer.propTypes = {
  // router
  match: PropTypes.object.isRequired,
};

export default withRouter(connect(mapStateToProps)(StockCountContainer));
