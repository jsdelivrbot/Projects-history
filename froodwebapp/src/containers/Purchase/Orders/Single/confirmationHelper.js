import React from 'react';
import { startCase } from 'lodash';
import { Icon } from 'antd';
import diff from 'object-diff';
import {
  confirmItem,
  confirmNewItem,
  confirmDeleteItem
} from './confirmationHelper.scss';

export default (formValues, initialValues) => {
  // order changes
  const orderChanges = diff(formValues, initialValues);

  const modalData = Object.getOwnPropertyNames(orderChanges).map(propName =>
    propName !== 'details' &&
    <p key={ propName } className={ confirmItem }>
      <Icon type="check" />
      { `${startCase(propName)} changed from ` }
      <span>{ `${orderChanges[propName]} ` }</span>
      to
      <span>{ ` ${formValues[propName]}` }</span>
    </p>
  );

  // if added new skuDetails or changed existing skuDetails
  formValues.details.forEach((skuDetail, index) => {
    if (skuDetail.deleted) return;

    const initialSkuDetail = initialValues.details[index];

    if (initialSkuDetail) { // if its not a new item
      const differ = diff(skuDetail, initialValues.details[index]);
      Object.getOwnPropertyNames(differ).map(propName =>
        modalData.push(
          <p key={ propName } className={ confirmItem }>
            <Icon type="check" />
            { `${skuDetail.name} ${startCase(propName)} changed from ` }
            <span>{ `${differ[propName]} ` }</span>
            to
            <span>{ ` ${skuDetail[propName]}` }</span>
          </p>
        )
      );
    } else { // if a new item
      modalData.push(
        <p key={ skuDetail.name } className={ confirmNewItem }>
          <Icon type="check" />
          { `Added ${skuDetail.name} with Qty ` }
          <span>{ `${skuDetail.qty} ` }</span>
          { Number(skuDetail.discount) !== 0 &&
            <span>
              <span>and Discount</span>
              <span>{ ` ${skuDetail.discount}` }</span>
            </span>
          }
        </p>
      );
    }
  });

  // if removed skuDetails
  formValues.details.filter(sku => sku.deleted === true && sku.new !== true)
    .forEach(skuDetail =>
      modalData.push(
        <p key={ skuDetail.name } className={ confirmDeleteItem }>
          <Icon type="check" />
        Removed
          <span>{ ` ${skuDetail.name}` }</span>
        </p>
      )
    );

  return modalData;
};
