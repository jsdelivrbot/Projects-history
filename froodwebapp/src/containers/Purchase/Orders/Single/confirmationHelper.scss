@import '~styles/colors.scss';

.confirmItem {
  color: $main-black;
  margin-bottom: 1rem;
  i {
    margin-right: 1rem;
    color: $main-green;
  }
  span {
    color: $main-red;
  }
  span:last-child {
    color: $main-green;
  }
}

.confirmNewItem {
  @extend .confirmItem;
  span {
    color: $main-green;
  }
  span:last-child {
    span:first-child {
      color: $main-black;
    }
  }
}

.confirmDeleteItem {
  @extend .confirmItem;
  i {
    color: $main-red;
  }
  span {
    color: $main-red;
  }
}
