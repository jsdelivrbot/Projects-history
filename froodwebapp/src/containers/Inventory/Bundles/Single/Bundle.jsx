import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { PageHeader } from 'components';
import { Tabs, ConnectedMainTab } from './Tabs';

export class Bundle extends Component {
  constructor(props) {
    super(props);
    const bundleId = props.match.params.id;
    const isNewBundle = bundleId === 'new';

    this.state = {
      bundleId,
      isNewBundle
    };
  }

  componentWillReceiveProps(nextProps) {
    const bundleId = nextProps.match.params.id;
    const isNewBundle = bundleId === 'new';

    this.setState({
      bundleId,
      isNewBundle
    });
  }

  render() {
    const {
      bundleId,
      isNewBundle
    } = this.state;

    return (
      <div>
        { isNewBundle &&
          <PageHeader
            bigText="Item Bundles"
          />
        }
        { isNewBundle &&
          <ConnectedMainTab
            isNewBundle={ isNewBundle }
          />
        }
        { !isNewBundle &&
          <Tabs
            bundleId={ bundleId }
            isNewBundle={ isNewBundle }
          />
        }
      </div>
    );
  }
}

Bundle.propTypes = {
  // router
  match: PropTypes.object.isRequired,
};

export default withRouter(Bundle);
