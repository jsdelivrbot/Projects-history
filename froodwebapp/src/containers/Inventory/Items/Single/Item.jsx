import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { PageHeader } from 'components';
import { Tabs, ConnectedMainTab } from './Tabs';

export class Item extends Component {
  constructor(props) {
    super(props);
    const itemId = props.match.params.id;
    const isNewSkuItem = itemId === 'new';

    this.state = {
      itemId,
      isNewSkuItem
    };
  }

  componentWillReceiveProps(nextProps) {
    const itemId = nextProps.match.params.id;
    const isNewSkuItem = itemId === 'new';

    this.setState({
      itemId,
      isNewSkuItem
    });
  }

  render() {
    const {
      itemId,
      isNewSkuItem
    } = this.state;

    return (
      <div>
        { isNewSkuItem &&
          <PageHeader
            bigText="Product Details"
            smallText="Enter your product name, type and other details to setup your product"
          />
        }
        { isNewSkuItem &&
          <ConnectedMainTab
            isNewSkuItem={ isNewSkuItem }
          />
        }
        { !isNewSkuItem &&
          <Tabs
            itemId={ itemId }
            isNewSkuItem={ isNewSkuItem }
          />
        }
      </div>
    );
  }
}

Item.propTypes = {
  match: PropTypes.object.isRequired,
};

export default withRouter(Item);
