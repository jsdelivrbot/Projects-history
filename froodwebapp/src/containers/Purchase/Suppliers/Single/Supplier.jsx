import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { PageHeader } from 'components';
import { Tabs, ConnectedMainTab } from './Tabs';

export class Supplier extends Component {
  constructor(props) {
    super(props);
    const supplierId = props.match.params.id;
    const isNewSupplier = supplierId === 'new';

    this.state = {
      supplierId,
      isNewSupplier
    };
  }

  componentWillReceiveProps(nextProps) {
    const supplierId = nextProps.match.params.id;
    const isNewSupplier = supplierId === 'new';

    this.setState({
      supplierId,
      isNewSupplier
    });
  }

  render() {
    const {
      supplierId,
      isNewSupplier
    } = this.state;

    return (
      <div>
        { isNewSupplier &&
          <PageHeader
            bigText="Product Supplier"
            smallText="Create new supplier for your products"
          />
        }
        { isNewSupplier &&
          <ConnectedMainTab
            supplierId={ supplierId }
            isNewSupplier={ isNewSupplier }
          />
        }
        { !isNewSupplier &&
          <Tabs
            supplierId={ supplierId }
            isNewSupplier={ isNewSupplier }
          />
        }
      </div>
    );
  }
}

Supplier.propTypes = {
  // router
  match: PropTypes.object.isRequired
};

export default withRouter(Supplier);
