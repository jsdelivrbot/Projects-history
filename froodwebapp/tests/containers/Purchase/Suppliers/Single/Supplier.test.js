import { Supplier } from 'containers/Purchase/Suppliers/Single/Supplier';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('Supplier', () => {
  const props = {
    match: {
      params: {
        id: 1
      }
    },
    location: {
      state: 'state'
    }
  };

  it('renders the Supplier component', () => {
    const component = shallow(<Supplier { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of Supplier', () => {
    const component = shallow(<Supplier { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
