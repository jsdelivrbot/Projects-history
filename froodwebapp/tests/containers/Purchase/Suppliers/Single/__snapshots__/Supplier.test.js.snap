// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supplier capturing Snapshot of Supplier 1`] = `
<div>
  <withRouter(Connect(PurchaseSupplierTabs))
    isNewSupplier={false}
    supplierId={1}
  />
</div>
`;
